commit 6ec9cd3096aff7a406cbb8d6830a81302e34527c
Author: Benjamin Gilbert <bgilbert@backtick.net>
Date:   Sat Dec 24 20:18:46 2022 -0500

    meson: allow disabling DWrite
    
    When building with MinGW-w64, DWrite introduces a libstdc++ dependency
    which may not be desired.

diff --git a/meson.build b/meson.build
index 180cfacf514b..756cfcd6ce33 100644
--- a/meson.build
+++ b/meson.build
@@ -506,13 +506,13 @@ if host_machine.system() == 'windows'
   ]
 
   cpp_compiler = meson.get_compiler('cpp')
-  d2d_dep = cpp_compiler.find_library('d2d1', required: false)
-  dwrite_dep = cpp_compiler.find_library('dwrite', required: false)
+  d2d_dep = cpp_compiler.find_library('d2d1', required: get_option('dwrite'))
+  dwrite_dep = cpp_compiler.find_library('dwrite', required: get_option('dwrite'))
   d2d_header = cpp_compiler.has_header('d2d1.h')
   d2d_3_header = cpp_compiler.has_header('d2d1_3.h')
   dwrite_header = cpp_compiler.has_header('dwrite.h')
   dwrite_3_header = cpp_compiler.has_header('dwrite_3.h')
-  wincodec_dep = cpp_compiler.find_library('windowscodecs', required: false)
+  wincodec_dep = cpp_compiler.find_library('windowscodecs', required: get_option('dwrite'))
   wincodec_header = cpp_compiler.has_header('wincodec.h')
 
   if d2d_dep.found() and dwrite_dep.found() and d2d_header and dwrite_header and wincodec_dep.found() and wincodec_header
diff --git a/meson_options.txt b/meson_options.txt
index ebc65c77e14b..1d4d018c42da 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,4 +1,5 @@
 # Cairo font backends
+option('dwrite', type : 'feature', value : 'auto')
 option('fontconfig', type : 'feature', value : 'auto')
 option('freetype', type : 'feature', value : 'auto')
 
